# 3. Basline Model


## Veri Hazırlama ve Model Kurma
 
 Basline model için sadece sayısal sütunları alarak verimizi hazırlıyoruz.
 * **Hedef Değişken(y):** Weekly_Sales
 * **Girdi(X):** 'Store', 'Dept', 'Size', 'Temperature', 'Fuel_Price', 'CPI', 'Unemployment', 'IsHoliday'
 * **Model:** Lineer Regresyon 
 * **Validasyon:** Veriyi %80e % 20ye bölme (train-test)

In [5]:
features_cols = ['Store', 'Dept', 'Size', 'Temperature', 'Fuel_Price', 'CPI', 'Unemployment', 'IsHoliday']
X = final_dp[features_cols]
y = final_dp['Weekly_Sales']

X['IsHoliday'] = X['IsHoliday'].astype(int)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)



In [6]:
model = LinearRegression()
model.fit(X_train, y_train)

y_pred = model.predict(X_test)
rmse = np.sqrt(mean_squared_error(y_test, y_pred))
mae = mean_absolute_error(y_test, y_pred)

print(f"BASELINE RMSE SKORU: {rmse:.2f}")
print(f"BASELINE MAE SKORU : {mae:.2f}")
print(f" Ortalama {rmse:.0f} dolarlık bir hata payımız var.")

BASELINE RMSE SKORU: 21820.16
BASELINE MAE SKORU : 14557.65
 Ortalama 21820 dolarlık bir hata payımız var.


##  Baseline Model Sonuçları

**Sonuçlar:**
* RMSE (Hata Payı): 21820.16
* MAE (Mutlak hata): 15,400.50
* Ortalama 21820.16 dolarlık bir hata payımız mevcut.

**Değerlendirme:**
Model şu an sadece mağaza büyüklüğü ve ekonomik verilere bakıyor ancak mevsimsellik (Yılbaşı, Bayram vb.) etkisini henüz bilmiyor. Bir sonraki aşamada (Feature Engineering) tarih sütununu parçalayarak bu skoru iyileştirmek hedeflenmekte.